src/Passes/site-packages/TasksCI/toolchain.py

```diff
@@ -2,31 +2,113 @@
 # Licensed under the MIT License.
 
 import shutil
+import os
 
+extensions = [""]
+try:
+    extensions = [x.lower() for x in os.getenv('PATHEXT').split(";")]
+except:  # noqa: E722
+    pass
 
-def discover_formatter() -> str:
+
+def discover_formatter_impl(ext: str) -> str:
     """
     Finds the clang-format executable
     """
-    return shutil.which("clang-format")
+    ret = shutil.which("clang-format{}".format(ext))
+    if ret is not None:
+        return ret
 
+    for v in ["11", "10", "12"]:
+        ret = shutil.which("clang-format-{}{}".format(v, ext))
+        if ret is not None:
+            return ret
 
-def discover_tidy() -> str:
+    return None
+
+
+def discover_formatter() -> str:
+    for ext in extensions:
+        ret = discover_formatter_impl(ext)
+        if ret is not None:
+            return ret
+
+    raise BaseException("Clang format not found")
+
+
+def discover_tidy_impl(ext: str) -> str:
     """
     Finds the clang-tidy executable
     """
-    return shutil.which("clang-tidy")
+    ret = shutil.which("clang-tidy")
+    if ret is not None:
+        return ret
 
+    for v in ["11", "11.0", "10", "10.0", "12", "12.0"]:
+        ret = shutil.which("clang-tidy-{}{}".format(v, ext))
+        if ret is not None:
+            return ret
 
-def discover_cmake() -> str:
+    possible_commands = [
+        "/usr/bin/clang-tidy-10",
+        "/usr/bin/clang-tidy-11",
+        "/usr/local/bin/clang-tidy-10",
+        "/usr/local/bin/clang-tidy-11",
+    ]
+
+    for c in possible_commands:
+        if os.path.exists(c):
+            return c
+
+    return None
+
+
+def discover_tidy() -> str:
+    for ext in extensions:
+        ret = discover_tidy_impl(ext)
+        if ret is not None:
+            return ret
+
+    raise BaseException("Clang tidy not found")
+
+
+def discover_cmake_impl(ext: str) -> str:
     """
     Finds the cmake executable
     """
-    return shutil.which("cmake")
+    ret = shutil.which("cmake{}".format(ext))
 
+    if ret is not None:
```

Comment:
This pattern seems a little odd... if `ret` is not `None` them we return it, but if it is `None` then we explicitly return `None`. Could this just be `return ret`?

Reply:
This is simply to follow the pattern of the previous function. The function attempts to find the command and if it succeeds, it returns. Most likely though, as we (eventually) get the Windows up running, we may need to add more tests. The significance of reaching the end of the function is that we have failed at finding `cmake` and we explicitly return `None` to make it clear that `None` is used as a error signal (see for instance the clang-tidy search). I am happy to simplify the function if you prefer that or add a placeholder comment say something like `# Test more locations`

Reply:
Ah, I see. That makes sense, thanks for explaining!